tests/integration/static_ip_address_test.py

```diff
@@ -0,0 +1,38 @@
+#
+# Copyright 2018 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+
+import copy
+
+from libnmstate import netapplier
+
+from .testlib import assertlib
+from .testlib import statelib
+from .testlib.statelib import INTERFACES
+
+
+def test_add_static_ipv4():
+    desired_state = statelib.show_only(('eth1',))
```

Comment:
IMHO the test should work without getting the current state of eth1 but specify the desired state indepentently and then check if it is properly applied. Otherwise the test condition depends on the previous state of the test setup which seems to be wrong.

Reply:
I think this is safer as it does not assume anything about what exists now.
My main concern is when we add other properties which may be a must: I do not want to revisit the test. In this solution, we get the template from what is supported now by nmstate and edit it.

We may want to add tests that check specific completion of data (when we provide only minimum data and expect nmstate to merge it with current), but I think that is not the same test... at least, it is a more messy one.

I agree we need not to drag problems or states between tests. For that I plan another change to be executed on every executed test, I will post it after we finish this initial one.

Reply:
I just added a second test, hope this is now ok.